contracts/DependencyAudit.cdc

```diff
@@ -0,0 +1,115 @@
+import "MigrationContractStaging"
+
+// This contract ist is used by the FVM calling the `checkDependencies` function from a function of the same name and singature in the FlowServiceAccount contract,
+// at the end of every transaction.
+// The `dependenciesAddresses` and `dependenciesNames` will be all the dependencies needded to run that transaction.
+//
+// The `checkDependencies` function will check if any of the dependencies are not staged in the MigrationContractStaging contract.
+// If any of the dependencies are not staged, the function will emit an event with the unstaged dependencies, or panic if `panicOnUnstaged` is set to true.
+access(all) contract DependencyAudit {
+
+    access(all) let AdministratorStoragePath: StoragePath
+
+    // The system addresses have contracts that will not be stages via the migration contract so we exclude them from the dependency chekcs
+    access(self) var excludedAddresses: {Address: Bool}
+
+    access(all) var panicOnUnstaged: Bool
+
+    access(all) event UnstagedDependencies(dependenciesAddresses: [Address], dependenciesNames: [String])
+
+    access(all) event PanicOnUnstagedDependenciesChanged(shouldPanic: Bool)
+
+    // checkDependencies is called from the FlowServiceAccount contract
+    access(self) fun checkDependencies(_ dependenciesAddresses: [Address], _ dependenciesNames: [String], _ authorizers: [Address]) {
```

Comment:
Looks like this param is unused, can this be removed?

Reply:
This is injected by the FVM. I added this incase we need to do some special handling according to authorizers.